pkgdb/src/lock-flake-installable.cc

```diff
@@ -0,0 +1,493 @@
+
+      if ( licenseIdValue != licenseValue->attrs->end()
+           && licenseIdValue->value->type() == nix::ValueType::nString )
+        {
+          return std::string( licenseIdValue->value->str() );
+        };
+    }
+
+  return std::nullopt;
+}
+
+LockedInstallable
+lockFlakeInstallable( const nix::ref<nix::EvalState> & state,
+                      const std::string &              system,
+                      const std::string &              installableStr )
+{
+  debugLog( nix::fmt( "original installable: %s", installableStr ) );
+
+  auto parsed = parseInstallable( installableStr );
+
+  nix::FlakeRef            flakeRef            = std::get<0>( parsed );
+  std::string              fragment            = std::get<1>( parsed );
+  nix::ExtendedOutputsSpec extendedOutputsSpec = std::get<2>( parsed );
+
+  debugLog(
+    nix::fmt( "original flakeRef: '%s'", flakeRef.to_string().c_str() ) );
+  debugLog( nix::fmt( "original fragment: '%s'", fragment ) );
+  debugLog( nix::fmt( "original extendedOutputsSpec: '%s'",
+                      extendedOutputsSpec.to_string() ) );
+
+  auto lockFlags = nix::flake::LockFlags {
+    .recreateLockFile      = false,
+    .updateLockFile        = false,
+    .writeLockFile         = false,
+    .useRegistries         = false,
+    .applyNixConfig        = false,
+    .allowUnlocked         = true,
+    .commitLockFile        = false,
+    .referenceLockFilePath = std::nullopt,
+    .outputLockFilePath    = std::nullopt,
+    .inputOverrides        = std::map<nix::flake::InputPath, nix::FlakeRef> {},
+    .inputUpdates          = std::set<nix::flake::InputPath> {}
+  };
+
+  nix::InstallableFlake installable = nix::InstallableFlake(
+    // The `cmd` argument is only used in nix to raise an error
+    // if `--arg` was used in the same command.
+    // The argument is never stored on the `InstallableFlake` struct
+    // or referenced outside of the constructor.
+    // We can safely pass a nullptr here, as the constructor performs a null
+    // check before dereferencing the arguement:
+    // <https://github.com/NixOS/nix/blob/509be0e77aacd8afcf419526620994cbbbe3708a/src/libcmd/installable-flake.cc#L86-L87>
+    static_cast<nix::SourceExprCommand *>( nullptr ),
+    state,
+    std::move( flakeRef ),
+    fragment,
+    extendedOutputsSpec,
+    // Defaults from nix:
+    // <https://github.com/NixOS/nix/blob/142e566adbce587a5ed97d1648a26352f0608ec5/src/libcmd/installables.cc#L231>
+    nix::Strings {
+      "packages." + system + ".default",
+      "defaultPackage." + system,
+    },
+    // Defaults from nix:
+    // <https://github.com/NixOS/nix/blob/142e566adbce587a5ed97d1648a26352f0608ec5/src/libcmd/installables.cc#L236>
+    nix::Strings {
+      "packages." + system + ".",
+      "legacyPackages." + system + ".",
+    },
+    lockFlags );
+
+  debugLog(
+    nix::fmt( "locked installable: '%s'", installable.what().c_str() ) );
+
```

Comment:
nonblocking: seems like this is a bit misleading and should really be after the `getLockedFlake()` call
```suggestion
```